app/fetchers/playlist.js

```diff
@@ -24,19 +24,34 @@ class PlaylistFetcher {
     if (typeof (channelPageDataResponse) === 'undefined') {
       channelPageDataResponse = response.data[1].response
     }
-    const channelMetaData = channelPageDataResponse.metadata.channelMetadataRenderer
-    const channelName = channelMetaData.title
-    const channelId = channelMetaData.externalId
+    if (typeof (channelPageDataResponse.alerts) !== 'undefined') {
+      return {
+        alertMessage: channelPageDataResponse.alerts[0].alertRenderer.text.simpleText
+      }
+    }
+    let channelName
+    let channelMetaData
+    let channelId
+    if ('metadata' in channelPageDataResponse) {
+      channelMetaData = channelPageDataResponse.metadata.channelMetadataRenderer
+      channelName = channelMetaData.title
+      channelId = channelMetaData.externalId
+    }
     const ytGrabHelp = helper.create(httpAgent)
 
     const channelInfo = {
       channelId: channelId,
       channelName: channelName,
       channelUrl: `https://www.youtube.com/channel/${channelId}`
     }
-
-    const playlistData = channelPageDataResponse.contents.twoColumnBrowseResultsRenderer.tabs[2].tabRenderer.content.sectionListRenderer.contents[0].itemSectionRenderer.contents[0].gridRenderer
-
+    let playlistData
+    const playlistTab = channelPageDataResponse.contents.twoColumnBrowseResultsRenderer.tabs.filter(e => {
+      return e.tabRenderer !== undefined && e.tabRenderer.title === 'Playlists'
+    })[0]
```

Comment:
instead of `.filter()[0]`, you can use `.find()` which returns the first match or `undefined` if there are no matches.

You could use the optional chaining operator to shorten the filter condition if you want to, as that was added in es2020 which the eslint config allows.

Also is there a way to do this without using a localised title? yt-channel-info currently hardcodes en-US as the language, however if that were to change in the future then this check would fail for other languages.

Reply:
I was actually planning on doing this in the next PR but I might as well do it now.